actinia_stac_plugin/core/stac.py

```diff
@@ -0,0 +1,265 @@
+    
+    # Caching JSON from the STAC collection
+    stac_json_collection =  requests.get(stac_root)
+    redis_actinia_interface.create(stac_unique_id,stac_json_collection)
+    
+    # Verifying the existance of the instances - Adding the item to the Default List
+    list_instances_exist = redis_actinia_interface.exists("stac_instances")
+    if not list_instances_exist:
+        defaultInstance() 
+    
+    stac_instance_exist= redis_actinia_interface.exists(stac_instance_id)
+
+    if not stac_instance_exist:
+        redis_actinia_interface.create(stac_instance_id,{})
+
+    defaultJson = redis_actinia_interface.read(stac_instance_id)
+
+    instances_list = redis_actinia_interface.read("stac_instances")
+
+    instances_list[stac_instance_id] = {
+       "path": "stac."+ stac_instance_id+".rastercube.<stac_collection_id>"
+    }
+
+    defaultJson[stac_unique_id] = {
+        'root': stac_root,
+        'href': "api/v1/stac/collections/" + stac_unique_id
+    }
+
+    list_of_instances_updated = redis_actinia_interface.update("stac_instances",instances_list)
+    instance_updated = redis_actinia_interface.update(stac_instance_id,defaultJson)
+
+    if instance_updated and list_of_instances_updated:
+        response = {
+            "message": "The STAC Collection has been added successfully",
+            "StacCatalogs": redis_actinia_interface.read(stac_instance_id)
+        }
+    else: 
+        response = {"message": "Something went wrong, please check the stac-instance-id , stac-url or stac-collection-id given"}
+        
+    return response
+
+def collectionValidation(url: str) -> bool:
+    """
+        Verify that the URL provided belong to a STAC endpoint
+    """
+    stac = stac_validator.StacValidate(url)
+    stac.run()
+    valid = stac.message[0]['valid_stac']
+    type = stac.message[0]['asset_type']
+    if valid and type == "COLLECTION":
+         return True
+    else: 
+        return False
+
+def addStacValidator(json):
+    """
+        The function validate the inputs syntax and STAC validity  
+        Input:
+            - json - JSON array with the Instance ID , Collection ID and STAC URL 
+    """
+    stac_instance_id = 'stac-instance-id' in json
+    stac_collecion_id = 'stac-collection-id' in json
+    stac_root = 'stac-url' in json
+    msg = {}
+    if stac_instance_id and stac_collecion_id and stac_root:
+        root_validation = collectionValidation( json['stac-url'])
+        collection_validation = (re.match('^[a-zA-Z0-9-_]*$',json['stac-collection-id']))
+        instance_validation = (re.match('^[a-zA-Z0-9-_]*$',json['stac-instance-id']))
+        
+        if root_validation and instance_validation and collection_validation:
+            return addStac2User(json)
+        elif not root_validation:
+            msg["Error-root"] = {"message": "Please check the URL provided (Should be a STAC Catalog) as well as the name given (no spaces or undercore characters)."}
```

Comment:
But aren't underscores allowed now?

Reply:
Ah - I confused it with `collection_validation`. But does this mean that we don't support stac URLs which contain an underscore? Or does it simply appear in this response but would work?